### ISSUE TITLE
Unable to correctly set the locale on Fedora-based distributions.

### ISSUE THREAD

**Issue by peter-777** [microsoft/WSL#11605]
### Windows Version

Microsoft Windows [Version 10.0.22631.3593]

### WSL Version

2.1.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.146.1-2

### Distro Version

Fedora 40 and  Oracle Linux 9.1

### Repro Steps

1. Set up a Fedora-based distribution such as Oracle Linux.
2. Enable systemd.
3. Execute the following commands:
```
sudo dnf install -y langpacks-ja
sudo localectl set-locale ja_JP.utf8
```
4. Log in again and execute a non-existent command.

### Expected Behavior

```
$ aaa
-bash: aaa: コマンドが見つかりません
```

### Actual Behavior

```
$ aaa
-bash: aaa: command not found
```

### Diagnostic Logs

Here are the findings from the experiment:

1.  By adding `printenv LANG` to the beginning of `/etc/profile` to check the variable, it shows en_US.UTF-8.

2. Strangely, when setting LANG in `/etc/default/locale`, the content is correctly reflected. If the file does not exist, it defaults to en_US.UTF-8.

Workaround:
Creating a symbolic link from /etc/locale.conf to /etc/default/locale resolves the issue.

**Comment by peter-777** [comment]
[WslLogs-2024-05-22_18-16-27.zip](https://github.com/microsoft/WSL/files/15401147/WslLogs-2024-05-22_18-16-27.zip)

**Comment by OneBlue** [comment]
@peter-777: What's the locale setting in Windows ? 

WSL inherits that settings by default

**Comment by peter-777** [comment]
@OneBlue The locale setting for Windows is set to Japanese.

**Comment by OneBlue** [comment]
Interesting. Is systemd enabled in your distribution ? Can you share the output of `cat /etc/wsl.conf` ?

**Comment by peter-777** [comment]
Yes, systemd is enabled in my distribution. Here is the output of cat /etc/wsl.conf:

```
[boot]
systemd = True
command = "mount --make-rshared /"
[user]
default = myuser
```

**Comment by OneBlue** [comment]
Ok thank you @peter-777. Also, can you share how you installed that distribution ?

**Comment by peter-777** [comment]
@OneBlue 
I have installed two distributions. Fedora is set as the default in WSL.
One of them is extracted from a Fedora container image and imported into WSL as a rootfs.

`wsl --import fedora-40 .\fedora-40\ .\fedora-40-rootfs.tar.gz`

The second one was to verify if the same issue occurs with a commonly available Fedora-based distribution. To do this, I installed Oracle Linux using the following command:

`wsl --install OracleLinux_9_1`
